src/main/java/lab2/Lab2.java

```diff
@@ -0,0 +1,199 @@
+classes:
+.send(request, handler) -	Sends a synchronous request
+.sendAsync(request, handler)- Sends an asynchronous request
+ */
+
+import java.net.http.HttpRequest;
+
+/*
+purpose: represents an HTTP response from a server. It contains status code, headers, and body.
+classes:
+
+HttpClient - Main class used to send requests and receive responses. Can be configured with options like redirect policy, proxy, and SSL.
+HttpRequest	- Represents an HTTP request. Built using a builder pattern (HttpRequest.newBuilder()).
+HttpResponse<T>	- Represents an HTTP response, with generic type T for the response body (e.g., String, InputStream).
+HttpRequest.BodyPublishers - Contains static methods to create request bodies (e.g., ofString(), ofFile()).
+HttpResponse.BodyHandlers -	Contains static methods to process response bodies (e.g., ofString(), ofInputStream()).
+WebSocket - Provides support for WebSocket communication (real-time, full-duplex).
+HttpHeaders - Represents HTTP headers (used in requests and responses).
+HttpTimeoutException - Exception thrown when a request times out.
+ */
+import java.net.http.HttpResponse;
+
+
+//method: AvatarGenerator
+//type: user-defined class
+//purpose: generate and display random avatars from the internet using the DiceBear API.
+
+void main() {
+    try {
+        var avatarStream = getRandomAvatarStream();
+        showAvatar(avatarStream);
+    } catch (IOException | InterruptedException e) {
+        e.printStackTrace();
+    }
+
+}
+
+//method: getRandomAvatarStream
+//type: public static
+//throws a IOException and InterruptedException when encountered certain runtime problems.
+InputStream getRandomAvatarStream() throws IOException, InterruptedException {
+
+
+    // Pick a random style
+    //Variable type: String[]
+    //list of styles
+    String[] styles = { "adventurer", "adventurer-neutral", "avataaars", "big-ears", "big-ears-neutral", "big-smile", "bottts", "croodles", "croodles-neutral", "fun-emoji", "icons", "identicon", "initials", "lorelei", "micah", "miniavs", "open-peeps", "personas", "pixel-art", "pixel-art-neutral" };
+    var style = styles[(int)(Math.random() * styles.length)];
+    //method: Math.random()
+    //type:  class method
+    //purpose: it will choose a random style from the  String[] styles
+
+    // Generate a random seed
+    // it generate random integer between 0 to 99999 to randomize the avatar
+    //method: Math.random()
+    var seed = (int)(Math.random() * 10000);
+
+    // Create an HTTP request for a random avatar
+    //method: URI.create
+    //the purpose of the url is to generate an avatar image by sending a request to the URL
+    //type: class method
+    var uri = URI.create("https://api.dicebear.com/9.x/%s/png?seed=%d".formatted(style, seed));
+    //method: HttpRequest.newBuilder(uri).build();
+    //type: instance method
+    //purpose: to build an http request
+    var request = HttpRequest.newBuilder(uri).build();
```

Comment:
There are two methods here: `newBuilder` and `build`.  The first is a class method, the second is an instance method.